src/Albany_ModelEvaluator.cpp

```diff
@@ -567,10 +567,19 @@ ModelEvaluator::create_hess_g_pp( int j, int l1, int l2 ) const
           << "l2 = " << l2
           << std::endl);
 
+  auto pl = app->getProblemPL()->sublist("Hessian").sublist(util::strint("Response",j)).sublist(util::strint("Parameter",l1));
+  bool HessVecProdBasedOp = pl.get("Reconstruct H_pp using Hessian-vector products",true);
+
   if (l1 < num_param_vecs) {
+    TEUCHOS_TEST_FOR_EXCEPTION(!HessVecProdBasedOp, std::logic_error, 
+            std::endl
+                << "Error!  Albany::ModelEvaluator::create_hess_g_pp():  "
+                << "Hessian pp operator for scalar response " << j << " and parameter " << l1 << " can only be reconstructed via Hessian-vector products"
+                << std::endl); 
```

Comment:
Just to be sure, the Hessian associated to a non-distributed parameter can only be created using the reconstruction via Hessian-vector products, is that right? If so, can we update the error message with  `<< " and (non-distributed) parameter " <<` or so.

Reply:
good point. I'll do that.